src/analysis/type_analysis.cpp

```diff
@@ -57,15 +57,13 @@ ConcreteCompilerType* NullTypeAnalysis::getTypeAtBlockEnd(InternedString name, C
 
 
 // Note: the behavior of this function must match irgenerator.cpp::unboxVar()
-static ConcreteCompilerType* unboxedType(ConcreteCompilerType* t) {
+static CompilerType* unboxedType(ConcreteCompilerType* t) {
     if (t == BOXED_BOOL)
         return BOOL;
-#if ENABLE_UNBOXED_VALUES
```

Comment:
I don't quite understand these code for now. But these change cause the error in #971. (I am reading these part in these days. Hope I could gather enough questions before 0.4 release. )